File: DSAlgorithms/Leetcode/c++/stacks/2390-remove_stars.cpp
class Solution {
public:
    string removeStars(string s) {
       std::stack<char> a_stack;
       for(int i = 0; i < s.size(); i++) 
       {
           if(s[i] != '*')
                a_stack.push(s[i]);
            else
                a_stack.pop();
       }
       std::string result = "";
       while(!a_stack.empty())
       {
           result += a_stack.top();
           a_stack.pop();
       }
       reverse(result.begin(), result.end());
       return result;
    }
};

File: dsa_algo/leetcode/ts/src/stacks/2390-remove_stars.ts
const removeStars = (s: string): string =>{
    const a_stack: string[] = [];
    for(let i = 0; i < s.length; i++)
    {
        if(s[i] === "*")
        {
            if(a_stack.length > 0)
                a_stack.pop();
        }
        else
            a_stack.push(s[i]);
    }
    let result = "";
    while(a_stack.length > 0)
    {
        result = a_stack[a_stack.length - 1] + result;
        a_stack.pop();
    }
    return result;
};